ietf/submit/utils.py

```diff
@@ -1039,3 +1129,120 @@ def remote_ip(request):
     else:
         remote_ip = request.META.get('REMOTE_ADDR', None)
     return remote_ip
+
+
+def _normalize_title(title):
+    if isinstance(title, str):
+        title = unidecode(title)  # replace unicode with best-match ascii
+    return normalize_text(title)  # normalize whitespace
+
+
+def process_submission_xml(submission):
+    """Validate and extract info from an uploaded submission"""
+    xml_path = staging_path(submission.name, submission.rev, '.xml')
+    xml_draft = XMLDraft(xml_path)
+
+    if submission.name != xml_draft.filename:
+        raise SubmissionError('XML draft filename disagrees with submission filename')
+    if submission.rev != xml_draft.revision:
+        raise SubmissionError('XML draft revision disagrees with submission revision')
+
+    authors = xml_draft.get_author_list()
+    for a in authors:
+        if not a['email']:
+            raise SubmissionError(f'Missing email address for author {a}')
+
+    author_emails = [a['email'].lower() for a in authors]
+    submitter = get_person_from_name_email(**submission.submitter_parsed())  # the ** expands dict into kwargs
+    if not any(
+            email.address.lower() in author_emails
+            for email in submitter.email_set.filter(active=True)
+    ):
+        raise SubmissionError(f'Submitter ({submitter}) is not one of the document authors')
+
+    # Fill in the submission data
+    submission.title = _normalize_title(xml_draft.get_title())
+    if not submission.title:
+        raise SubmissionError('Could not extract a valid title from the XML')
+    submission.authors = [
+        {key: auth[key] for key in ('name', 'email', 'affiliation', 'country')}
+        for auth in authors
+    ]
+    submission.xml_version = xml_draft.xml_version
+    submission.save()
+
+
+def process_submission_text(submission):
+    """Validate/extract data from the text version of a submitted draft
+
+    This assumes the draft was uploaded as XML and extracts data that is not
+    currently available directly from the XML. Additional processing, e.g. from
+    get_draft_meta(), would need to be added in order to support direct text
+    draft uploads.
+    """
+    text_path = staging_path(submission.name, submission.rev, '.txt')
+    text_draft = PlaintextDraft.from_file(text_path)
+
+    if submission.name != text_draft.filename:
+        raise SubmissionError('Text draft filename disagrees with submission filename')
+    if submission.rev != text_draft.revision:
+        raise SubmissionError('Text draft revision disagrees with submission revision')
+    if not _normalize_title(text_draft.get_title()):
```

Comment:
Check title with submission title

Reply:
There had been a check for consistency between .txt and .xml titles, so restoring that.